tests/func/test_data_cloud.py

```diff
@@ -308,6 +310,68 @@ def test_pull_git_imports(tmp_dir, dvc, scm, erepo):
     assert (tmp_dir / "new_dir" / "bar").read_text() == "bar"
 
 
+@pytest.mark.parametrize(
+    "erepo", [pytest.lazy_fixture("git_dir"), pytest.lazy_fixture("erepo_dir")]
+)
+def test_pull_git_imports_with_store(tmp_dir, dvc, scm, erepo, local_remote):
+    with erepo.chdir():
+        erepo.scm_gen({"dir": {"bar": "bar"}}, commit="second")
+        erepo.scm_gen("foo", "foo", commit="first")
+
+    dvc.imp(os.fspath(erepo), "foo", store=True)
+    dvc.imp(os.fspath(erepo), "dir", out="new_dir", store=True, rev="HEAD~")
+
+    assert dvc.pull()["fetched"] == 0
+    assert dvc.push() == 3
+
+    for item in ["foo", "new_dir", dvc.cache.local.cache_dir]:
+        remove(item)
+    os.makedirs(dvc.cache.local.cache_dir, exist_ok=True)
+    clean_repos()
+
+    # Close the repo since otherwise Windows holds references open which
+    # prevent deletion
+    erepo.close()
+    remove(erepo)
+
+    assert dvc.pull(force=True)["fetched"] == 2
+
+    assert (tmp_dir / "foo").exists()
+    assert (tmp_dir / "foo").read_text() == "foo"
+
+    assert (tmp_dir / "new_dir").exists()
+    assert (tmp_dir / "new_dir" / "bar").read_text() == "bar"
+
+
+def test_store_false_pull_error(tmp_dir, dvc, scm, local_remote, caplog):
+    tmp_dir.dvc_gen("foo", "Locally added", commit="add tracked file")
+
+    dvc.push()
+    clean(["foo"], dvc)
+    dvc.pull()
+
+    with open(tmp_dir / "foo.dvc", "a") as F:
+        F.write("  store: false\n")
+
+    dvc.push()
+    clean(["foo"], dvc)
+
+    missing_error = (
+        "Cache 'HashInfo(name='md5', "
+        "value='d089d68656330fd712dad63400c8710f', dir_info=None)' not found. "
+        "File 'foo' won't be created"
+    )
+    missing_error_hint = (
+        "The following files are marked as `store: false` so the remote was "
+        "not searched, but are not present in local cache:\nfoo"
```

Comment:
I guess this would need
```suggestion
        "The following outputs are marked as `store: false` so remote "
        "storage was disabled, and are not present in the cache:\nfoo"
```